x-pack/plugins/apm/server/routes/source_maps/schedule_source_map_migration.ts

```diff
@@ -0,0 +1,216 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { FleetStartContract } from '@kbn/fleet-plugin/server';
+import { FleetArtifactsClient } from '@kbn/fleet-plugin/server/services';
+import { TaskManagerSetupContract } from '@kbn/task-manager-plugin/server';
+import { CoreStart, Logger } from '@kbn/core/server';
+import { getApmArtifactClient } from '../fleet/source_maps';
+import { bulkCreateApmSourceMaps } from './bulk_create_apm_source_maps';
+import { APM_SOURCE_MAP_INDEX } from '../settings/apm_indices/get_apm_indices';
+import { ApmSourceMap } from './create_apm_source_map_index_template';
+import { APMPluginStartDependencies } from '../../types';
+import { createApmSourceMapIndexTemplate } from './create_apm_source_map_index_template';
+
+const PER_PAGE = 10;
+const TASK_ID = 'apm-source-map-migration-task-id';
+const TASK_TYPE = 'apm-source-map-migration-task';
+
+export async function scheduleSourceMapMigration({
+  coreStartPromise,
+  pluginStartPromise,
+  fleetStartPromise,
+  taskManager,
+  logger,
+}: {
+  coreStartPromise: Promise<CoreStart>;
+  pluginStartPromise: Promise<APMPluginStartDependencies>;
+  fleetStartPromise?: Promise<FleetStartContract>;
+  taskManager?: TaskManagerSetupContract;
+  logger: Logger;
+}) {
+  if (!taskManager) {
+    return;
+  }
+
+  logger.debug(`Register task "${TASK_TYPE}"`);
+
+  taskManager.registerTaskDefinitions({
+    [TASK_TYPE]: {
+      title: 'Migrate fleet source map artifacts',
+      description:
+        'Migrates fleet source map artifacts to `.apm-source-map` index',
+      timeout: '24h',
+      maxAttempts: 5,
+      maxConcurrency: 1,
+      createTaskRunner(context) {
+        // const { state } = context.taskInstance;
+        // logger.debug({ state });
+        return {
+          async run() {
+            try {
+              logger.debug(`Run task: "${TASK_TYPE}"`);
+              const coreStart = await coreStartPromise;
+              const internalESClient =
+                coreStart.elasticsearch.client.asInternalUser;
+
+              // ensure that the index template has been created before running migration
+              await createApmSourceMapIndexTemplate({
+                client: internalESClient,
+                logger,
+              });
+
+              const fleet = await fleetStartPromise;
+              if (fleet) {
+                await runFleetSourcemapArtifactsMigration({
+                  fleet,
+                  internalESClient,
+                  logger,
+                });
+              }
+
+              logger.debug(`Task completed: "${TASK_TYPE}"`);
+            } catch (e) {
+              logger.error(`Task failed: "${TASK_TYPE}"`);
+              logger.error(e);
```

Comment:
It looks like you're specifying `maxAttempts: 5` which indicates you would like to retry but by swallowing the error here, task manager will assume the task succeeded and not retry.

Reply:
Good point. I'll remove the try/catch and won't swallow any exceptions. Is that the recommended approach?